### PR TITLE
feat(modal): static method to overwrite styles

### PR DESCRIPTION
**Pull request by gcottrell1**
allows us to overwrite the styles so that we can remove default borders/backgrounds

image

### CODE REVIEW

**Comment by mattyclarkson** [comment]
Love the copy in the modal 😃

File: lib/index.ts [hunk 1]
```diff
@@ -4,4 +4,7 @@ export { setup };
 import Backend from '@lib/Backend';
 export { Backend };
 
+import Modal from '@lib/screen/Modal';
+export { Modal };
```
Comment (mattyclarkson) [inline lib/index.ts]: You prolly don't want to expose the component outside of the bundle. You could do this my exporting a getter/setter for the property like so:

```typescript
export { getStyle as modalScreenGetStyle, setStyle as modalScreenSetStyle } from '@lib/screen/Modal';
```

You could implement them like so:

```typescript
let style: StyleProp<ViewStyle> = styles.modal;

function getStyle(): StyleProp<ViewStyle> {
  return style;
}

function getStyle(value : StyleProp<ViewStyle>): void {
 style = value;
}
```

The default value for `style` (`styles.modal`) is just what was there before. Doing `setStyle` truely overrides the default styles for the modal. You can customise all the style, including the min/max width.

It also makes it easy for uses to extend the base style like so:

```typescript
import { modalScreenGetStyle, modalScreenSetStyle } from '@ef-carbon/react-native-navigation-backend-react-navigation';

const style = StyleSheet.flatten([modalScreenGetStyle()]);
style.padding = 20;
modalScreenSetStyle(style);
```

What do you think?

Comment (gcottrell1) [reply lib/index.ts]: Ye I think that is a much better way to override the default styles. I'll implement this now.

File: lib/screen/Modal/styles.ts [hunk 2]
```diff
@@ -15,10 +15,6 @@ export default StyleSheet.create({
     })()),
   },
   modal: {
-    backgroundColor: colour.grey.lightest,
-    borderColor: colour.grey.normal,
-    borderWidth: border.width.normal,
-    borderRadius: border.radius.large,
```
Comment (mattyclarkson) [inline lib/screen/Modal/styles.ts]: I think leaving the default styling would help people get started with the backend. I feel having to force uses to call `modalScreenSetStyle` to have a default styling is a PITA.

Comment (gcottrell1) [reply lib/screen/Modal/styles.ts]: Woops, didn't meant to leave this removed, one problem I have noticed is that the defaults and the overrides are now tied and so the defaults will override values provided by the theme, so I think I'm going to separate them back out into two variables

File: lib/screen/Modal/styles.ts [hunk 3]
```diff
@@ -16,7 +16,7 @@ export default StyleSheet.create({
   },
   modal: {
     backgroundColor: colour.grey.lightest,
-    borderColor: colour.grey.normal,
+    borderColor: colour.grey.normal,
```
Comment (mattyclarkson) [inline lib/screen/Modal/styles.ts]: You've got some trailing whitespace here. I'm sure it'll get cleaned up at some point though.

File: lib/screen/Modal/Component.tsx [hunk 4]
```diff
@@ -41,23 +51,24 @@ export class ModalScreen extends React.PureComponent<IProps> {
...
     const safe = theme ? theme.padding.safe.all : padding.safe.all;
 
     return (
       <View style={[styles.container, safe]}>
-        <View style={[styles.modal, style]}>{node}</View>
+        <View style={[defaultStyle, overrideStyle, propStyle]}>{node}</View>
       </View>
     );
   }
+
```
Comment (mattyclarkson) [inline lib/screen/Modal/Component.tsx]: Unneeded new line here.

**Pull request merged**